src/utils.js

```diff
@@ -71,6 +71,28 @@ const utils = {
     return immutableData.every((item) => !item || item.isEmpty());
   },
 
+  flattenMap(data) {
+    return data.reduce(
+      (flattened, section, key) => flattened.set(key, section).merge(section),
+      Immutable.OrderedMap().asMutable(),
+    ).asImmutable();
+  },
+
+  isSectionHeader(item) {
+    return Immutable.Map.isMap(item) || Immutable.List.isList(item);
```

Comment:
Why are Map and List necessarily section headers? I don't think I understand how this method works.

Reply:
Yeah, I wasn't totally happy about this. In the `flattenMap` method, I'm setting the value of the section header items to be their respective section, in other words:
```javascript
Map {
  section1: Map {
    key1: val1,
    key2: val2
  },
  section2: Map {
    key3: val3,
    key4: val4
  }
}

// gets flattened to:

Map {
  section1: Map {
    key1: val1,
    key2: val2
  },
  key1: val1,
  key2: val2,
  section2: Map {
    key3: val3,
    key4: val4
  },
  key3: val3,
  key4: val4
}
```
One way, therefore, of identifying whether or not an item is a section header, is by querying if the item's value is a `Map` or a `List`. Do you think there's a better way?

Reply:
Do you happen to have a link to where the original ListView does something like this?

Reply:
`VirtualizedSectionList` does it like this: https://github.com/facebook/react-native/blob/master/Libraries/Lists/VirtualizedSectionList.js#L175-L224

For each index, it runs through the entire collection to work out if the index refers to a section header. I think this could be expensive, especially given that we are working with a `Map` which has keys rather than indexes so we'd have to convert to an indexed collection too.

Actually, I think I can use `stickyHeaderIndices`, as we have already computed these. Will put this in the next commit!